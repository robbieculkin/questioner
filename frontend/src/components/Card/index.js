import React from 'react';
import './index.scss';

import Entry from '../Entry';

const Card = ({ history, onTextEntry }) =>
  <div className='card'>
    <div className='card-item'>
        <div className='question'>
          {history[history.length - 1].text}
        </div>
        <Entry onTextEntry={onTextEntry} />
    </div>
  </div>

export default Card;