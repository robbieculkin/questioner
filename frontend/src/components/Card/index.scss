.card {
    background: #fff;
    padding: 50px;
    border-radius: 8px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);

    .card-item {
        text-align: center;

        .question {
            width: 100%;
            font-size: 24px;
            margin-bottom: 30px;
        }

        .response {
            display: flex;
            flex-direction: column;
            align-items: flex-start;
        }
    }
}