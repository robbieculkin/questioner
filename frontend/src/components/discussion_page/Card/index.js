import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

import Entry from '../Entry';

const Card = ({ history, onTextEntry }) =>
  <div className='card'>
    <div className='card-item'>
      <div className='question'>
        {history[history.length - 1].text}
      </div>
      <div className='response'>
        <Entry onTextEntry={onTextEntry} />
      </div>
    </div>
  </div>

Card.propTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string
    })
  ),
  onTextEntry: PropTypes.func
};

Card.defaultProps = {
  history: [{ text: '' }]
}

export default Card;