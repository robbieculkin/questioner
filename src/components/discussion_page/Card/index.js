import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

import Entry from '../Entry';

const process_html = text => {
  if (!text.includes('\\q'))
    return text;

  text = text.split(/\\q|"/)
  if (text.length == 4) {
    return (
      <div>
        <span>{text[0]}</span>
        <br/>
        <br/>
        <div className='quote-container'>
          <span>{text[1]}</span>
          <br/>
          <span className='quote-text'>
            <span className='accent-symbol'>"</span>
            {text[2]}
            <span className='accent-symbol'>"</span>
          </span>
          <br/>
          <span className='act-scene'>{text[3]}</span>
        </div>
      </div>
    );
  }
}

const Card = ({ history, onTextEntry, isReport }) =>
  <div className='card'>
    <div className='card-item'>
      {history.map(message =>
        <div className={`feed-item ${message.fromUser ? 'right' : 'left'}`} key={message.msgId}>
          <div className={`${message.fromUser ? 'response' : 'question'} ${message.msgId === history[history.length - 1].msgId || isReport ? '' : 'old-message'}`}>
            {process_html(message.text)}
          </div>
        </div>
      )}
      {isReport
        ? ''
        : <div className='response entry'>
            <Entry onTextEntry={onTextEntry} />
          </div>
      }
    </div>
  </div>

Card.propTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string
    })
  ),
  onTextEntry: PropTypes.func
};

Card.defaultProps = {
  history: [{ text: '' }]
}

export default Card;