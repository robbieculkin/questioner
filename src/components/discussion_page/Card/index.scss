.card {
  padding: 40px 0;
  width: 50%;

  .card-item {
    text-align: right;
    font-size: 28px;
    display: flex;
    flex-direction: column;
    align-items: flex-end;

    .old-message {
      transition: color 0.5s ease-in-out;
      color: #666;
      .quote-container {
        border-right: 4px solid rgba(46, 170, 77, 0.5) !important;
      }
    }

    .question {
      font-weight: 600;
      font-style: italic;
      margin-bottom: 30px;

      .quote-container {
        padding-right: 20px;
        border-right: 4px solid rgb(46, 170, 77);
        .accent-symbol {
          // color: rgb(46, 170, 77);
        }
        .quote-text {
          font-weight: 400;
        }
        .act-scene {

        }
      }
    }

    .response {
      margin-bottom: 80px;
      width: 650px;
      word-wrap: break-word;

      &.entry {
        height: 250px;
        margin-bottom: 5%;
      }
    }
  }
}