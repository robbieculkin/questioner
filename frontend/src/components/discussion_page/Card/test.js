import React from 'react';
import { shallow } from 'enzyme';
import Card from '.';

describe('Card', () => {
  it('should render correctly without props', () => {
    const cardProps = { history: [{ text: 'Text Here' }] };
    const component = shallow(<Card {...cardProps} />);
    expect(component).toMatchSnapshot();
  });
});
