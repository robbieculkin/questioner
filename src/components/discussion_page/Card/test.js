import React from 'react';
import { shallow } from 'enzyme';
import Card from '.';

describe('Card', () => {
  it('should render correctly without props', () => {
    const component = shallow(<Card />);
    expect(component).toMatchSnapshot();
  });
});
