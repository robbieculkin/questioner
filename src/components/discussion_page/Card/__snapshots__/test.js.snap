// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly without props 1`] = `
<div
  className="card"
>
  <div
    className="card-item"
  >
    <div
      className="question"
    />
    <div
      className="response"
    >
      <Entry />
    </div>
  </div>
</div>
`;
